docs/manual/src/internals/bindings_render_pipeline.md

```diff
@@ -0,0 +1,307 @@
+This metadata is then serialized into a buffer and stored as an exported symbol in the Rust library.
+The generated code will look something like this:
+
+```rust
+// Medata buffer for the `arithmetic::add()` function
+const UNIFFI_META_CONST_ARITHMETIC_FUNC_ADD: ::uniffi::MetadataBuffer = ::uniffi::MetadataBuffer::from_code(
+        ::uniffi::metadata::codes::FUNC, // Code for the item type
+    )
+    .concat_str("arithmetic") // module name
+    .concat_str("add") // function name
+    .concat_bool(false) // async?
+    .concat_value(2u8) // number of arguments
+    .concat_str("a") // 1st argument name
+    .concat(<u64 as ::uniffi::TypeId<crate::UniFfiTag>>::TYPE_ID_META) // 1st argument type
+    .concat_bool(false) // default value?
+    .concat_str("b") // 2nd argument name
+    .concat(<u64 as ::uniffi::TypeId<crate::UniFfiTag>>::TYPE_ID_META) // 2nd argument type
+    .concat_bool(false) // default value?
+    .concat(<Result<u64> as ::uniffi::TypeId<crate::UniFfiTag>>::TYPE_ID_META) // result type
+    .concat_long_str(""); // docstring
+
+// Serialize the metadata into a byte array and export the symbol.
+//  This can be evaluated at compile time because `UNIFFI_META_ARITHMETIC_FUNC_ADD` is `const`.
+#[no_mangle]
+#[doc(hidden)]
+pub static UNIFFI_META_ARITHMETIC_FUNC_ADD: [u8; UNIFFI_META_CONST_ARITHMETIC_FUNC_ADD
+    .size] = UNIFFI_META_CONST_ARITHMETIC_FUNC_ADD.into_array();
+```
+
+Notes:
+
+* UniFFI gets the type metadata types using the `TypeId::TYPE_ID_META`.
+  By using a trait and associated type rather than relying on the type identifier, we get the correct metadata even through type aliases.
+  For example, `Result<u64>` is a type alias with the error type omitted, but `<Result<u64> as ::uniffi::TypeId<crate::UniFfiTag>>::TYPE_ID_META>` includes metadata about the error type.
+* See [Lifting and Lowering](./lifting_and_lowering.md) for details on `UniFfiTag`.
+
+When a UniFFI bindgen command runs, it reads/deserializes the exported metadata symbols from the Rust library.
+Then the metadata is passed on to the next stage of the pipeline
+See `uniffi_bingen::macro_metadata` and `uniffi_meta::reader` for details.
+
+Here's how the metadata might look for the above function:
+```
+FnMetadata {
+    module_path: "arithmetic",
+    name: "add",
+    is_async: false,
+    inputs: [
+        FnParamMetadata {
+            name: "a",
+            ty: UInt64,
+            by_ref: false,
+            optional: false,
+            default: None,
+        },
+        FnParamMetadata {
+            name: "b",
+            ty: UInt64,
+            by_ref: false,
+            optional: false,
+            default: None,
+        },
+    ],
+    return_type: Some(
+        UInt64,
+    ),
+    throws: Some(
+        Enum {
+            module_path: "arithmetic",
+            name: "ArithmeticError",
+        },
+    ),
+    checksum: Some(
+        39863,
+    ),
+    docstring: None,
+},
+```
+
+
+## Metadata from the UDL file
```

Comment:
nit: Since there aren't details about this process (which I'm assuming is because pro-macros are favored over UDLs), you could remove this section, generalize the heading for the proc-macro metadata section, and add this as a note there.